geode-core/src/main/java/org/apache/geode/internal/net/BufferPool.java

```diff
@@ -43,8 +44,8 @@ public BufferPool(DMStats stats) {
   /**
    * A list of soft references to byte buffers.
    */
-  private final ConcurrentLinkedQueue<BBSoftReference> bufferQueue =
-      new ConcurrentLinkedQueue<>();
+  private final ConcurrentLinkedDeque<ByteBufferSoftReference> bufferQueue =
```

Comment:
What was the purpose in converting to a deque?

Reply:
Darrel asked the same question below. see the answer here: https://github.com/apache/geode/pull/3741#discussion_r297357593